crates/common/tedge_utils/src/file.rs

```diff
@@ -0,0 +1,53 @@
+use std::process::Command;
+
+pub fn create_directory_with_user_group(
+    grp_user: &str,
+    dirs: Vec<&str>,
+) -> Result<(), anyhow::Error> {
+    for directory in dirs {
+        match Command::new("install")
+            .args(["-g", grp_user, "-o", grp_user, "-m", "775", "-d", directory])
+            .output()
+        {
+            Ok(_) => {}
+            Err(e) => {
+                eprintln!(
+                    "failed to create the folder {} due to error {}",
+                    directory,
+                    e.to_string()
+                );
+                return Err(e.into());
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn create_file_with_user_group(grp_user: &str, files: Vec<&str>) -> Result<(), anyhow::Error> {
+    for file in files {
+        match Command::new("install")
+            .args([
+                "-g",
+                grp_user,
+                "-o",
+                grp_user,
+                "-m",
+                "644",
+                "/dev/null",
+                file,
+            ])
+            .output()
+        {
+            Ok(_) => {}
+            Err(e) => {
+                eprintln!(
+                    "failed to create the file {} due to error {}",
+                    file,
+                    e.to_string()
```

Comment:
Useless `.to_string()`.
```suggestion
                    e
```